AsyncDisplayKit/ASMapNode.mm

```diff
@@ -185,9 +191,23 @@ - (void)removeLiveMap
     _mapView = nil;
     _mapNode = nil;
   }
-  self.image = nil;
 }
 
+- (void)setAnnotations:(NSArray *)annotations
+{
+    ASDN::MutexLocker l(_propertyLock);
+    _annotations = [annotations copy];
+    if (annotations.count != _annotations.count) {
```

Comment:
This condition could never trigger, right?  Since you are setting _annotations = annotations above?  Either way, comparing the count is not the same as isEqual:, because you could have two entirely different sets of annotations that happen to have the same number of them.

If we have a live map, we probably shouldn't call setNeedsDisplay.  Since the user may scroll around the _mapView the snapshot will become out of date anyway, so I think we shouldn't spend resources fetching map tiles and rendering them while the live map is active.

If _mapView is non-nil /if and only if/ we have live map mode set, maybe we should just get rid of the boolean entirely and have - (BOOL)isLiveMap return (_mapView != nil);   It's a bit strange to check if (_mapView) here where some places check live map.